src/common/mfu_daos.c

```diff
@@ -293,10 +294,37 @@ int daos_parse_path(
 {
     struct duns_attr_t  dattr = {0};
     int                 rc;
+    char*               tmp_path1 = NULL;
+    char*               path_dirname = NULL;
+    char*               tmp_path2 = NULL;
+    char*               path_basename = NULL;
+    char*               tmp = NULL;
+
+    /* call duns_resolve_path on dirname. If basename does not
+     * exist then duns_resolve_path will fail even if dirname is a 
+     * UNS path */
+
+    /* get dirname */
+    tmp_path1 = strdup(path);
+    path_dirname = dirname(tmp_path1);
 
     /* Check if this path represents a daos pool and/or container. */
-    rc = duns_resolve_path(path, &dattr);
+    rc = duns_resolve_path(path_dirname, &dattr);
     if (rc == 0) {
+        /* if duns_resolve_path succeeds then concat basename to 
+         * da_rel_path */
+        tmp_path2 = strdup(path);
+        path_basename = basename(tmp_path2);
+        
+        /* make sure there is enough memory allocated to concat basename */
+        tmp = realloc(dattr.da_rel_path, path_len);
+        if (tmp == NULL) {
+            rc = ENOMEM;
+        }
+        dattr.da_rel_path = tmp;
+        strcat(dattr.da_rel_path, "/");
+        strcat(dattr.da_rel_path, path_basename);
```

Comment:
@dsikich If tmp==NULL here then it wouldn't be safe to call these strcat functions. And the caller expects a negative rc to be an actual error.
To remove the need for this extra error handling, maybe instead of calling realloc at all, we could just append basename directly below? (line 331) Either with strcat, asprintf, etc.

Reply:
@daltonbohning not sure what you mean, I don't think appending directly to da_rel_path is safe? Thats fine, I can return there.

Reply:
@dsikich What I mean is that instead of creating a copy of rel_path and appending the basename, and then doing this:
https://github.com/hpc/mpifileutils/blob/f6d2143d5dc669f002c4bffebc426df827771a71/src/common/mfu_daos.c#L331-L335
We could instead just directly append the basename to `path`.

Reply:
@daltonbohning copying over the path string seems to work ok with strncpy but using strcat causes memory corruption errors

Reply:
@daltonbohning I updated this again though, this way also seems to work but it requires using strncpy

Reply:
@dsikich 
> copying over the path string seems to work ok with strncpy but using strcat causes memory corruption errors

Oh, right. Because the string isn't null terminated (or zero'ed out).